agents/tools/filesystem.py

```diff
@@ -0,0 +1,61 @@
+import asyncio
+from pathlib import Path
+
+from pydantic import BaseModel, Field
+
+from beeai_framework.context import RunContext
+from beeai_framework.emitter import Emitter
+from beeai_framework.tools import StringToolOutput, Tool, ToolError, ToolRunOptions
+
+from utils import get_absolute_path
+
+
+class GetCWDToolInput(BaseModel):
+    pass
+
+
+class GetCWDTool(Tool[GetCWDToolInput, ToolRunOptions, StringToolOutput]):
+    name = "get_cwd"
```

Comment:
interesting; where is this tool coming from?

Reply:
What do you mean? Are you asking why is it needed? The current directory is purposefully passed in the prompt, but the models still call `pwd` using `run_shell_command` when they fail to provide absolute path to a tool that needs it.